File: scala-code/src/main/scala/com/springernature/slides/Slide6.scala
package com.springernature.slides

import java.io.{File, FileWriter}
import scala.io.Source

object Slide6 {

  def writeStringToFile(file: File, block: String): Unit = {
    val fileWriter = new FileWriter(file)
    fileWriter.write(block)
    fileWriter.close()
  }

  def writeToFile(file: File, block: => String): Unit = {
    if(file.isFile)
      writeStringToFile(file, block)
  }

  def main(args: Array[String]): Unit = {
    val file = new File("./text.txt")
    file.createNewFile()
    writeToFile(file, "Scala rocks! " * 1000)
  }
}

File: kotlin-code/src/main/kotlin/com/springernature/slides/Slide6.kt
package com.springernature.slides

import java.io.File

fun writeStringToFile(file: File, block: String) {
    file.writeText(block)
}

fun writeToFile(
    file: File,
    block: () -> String
) {
    if (file.isFile)
        writeStringToFile(file, block())
}

fun main() {
    val file = File("./text.txt")
    file.createNewFile()
    writeToFile(file) {
        "Kotlin rocks! ".repeat(1000)
    }
}
